tests/src/Integration/Action/UserRoleAddTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\UserRoleAddTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+use Drupal\Tests\rules\Integration\RulesUserIntegrationTestTrait;
+use Drupal\user\RoleInterface;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\UserRoleAdd
+ * @group rules_actions
+ */
+class UserRoleAddTest extends RulesEntityIntegrationTestBase {
+
+  use RulesUserIntegrationTestTrait;
+
+  /**
+   * The action that is being tested.
+   *
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->action = $this->actionManager->createInstance('rules_user_role_add');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary()
+   */
+  public function testSummary() {
+    $this->assertEquals('Adds roles to a particular user', $this->action->summary());
+  }
+
+  /**
+   * Tests adding of one role to user.
+   *
+   * @covers ::execute()
+   */
+  public function testAddOneRole() {
+    // Set-up a mock user.
+    $account = $this->getMockUser();
+    $account->expects($this->once())
+      ->method('addRole');
```

Comment:
We should also have a ->with() call here to make sure the addRole() method is called with the correct parameter.